source/exercises/mailroom-oo.rst

```diff
@@ -51,29 +52,64 @@ One of the hardest parts of OO design (particularly in Python) is to know how "l
 
 There are no hard and fast rules, but here are some guidelines:
 
-For this assignment it's OK to go with simple tuples. However, in order for the code to be more flexible in the future, for new "fields" were added to the donor object, it's probably better to use a more structured type, so you don't have to worry about changing the order or number of fields.
+For this assignment it's OK to go with simple tuples. However, in order for the code to be more flexible in the future, if new "fields" were added to the donor object, it's probably better to use a more structured type, so you don't have to worry about changing the order or number of fields.
 
 So now you have to think about using a dict or class? Again for flexibility, I think a dict is a bit easier -- you can add fields to it very easily. However, with a class, you can build some functionality in there, too. This is where encapsulation comes in -- for instance, one thing you might want to do is get the total of all donations a donor has made in the past. If you add a method to compute that (or a property!), then the rest of the code doesn't need to know how the donations are stored.
 
-So: if you want my hints :-) ...
+Consider ``data[0]`` vs ``data["first_name"]`` vs ``data.first_name`` -- which one is more readable? Keep in mind that another benefit of using OO for data encapsulation is ability of modern IDE to provide auto-completion, which reduces number of bugs and helps to produce code faster.
+
+Below are more detailed suggestions on breaking down your existing code into multiple modules that will be part of a single mailroom program.
+
+Donor Class
+^^^^^^^^^^^^
```

Comment:
RST allows pretty much any charactor to make headings -- but I've tried ot be consistent in these docs with:

`======`
then
`--------`
then 
`........`

so I think this level should be:
`..............`

Reply:
BTW -- are you building the docs?

running `make html` in the root dir of the repo should build them for you locally.

Reply:
ah good to know! will update heading.

no I did not run `make html`... I feel like you've mentioned this before and ended up doing it for me last time. I'll see if I can figure it out. (haven't worked with this type of docs before).

FYI I'm using Pycharm with a plugin to render these.